e2e/cache/cache.go

```diff
@@ -18,88 +20,290 @@ type cacheTests struct {
+			needImage:      false,
+			exit:           0,
+		},
+		{
+			name:           "list help",
+			options:        []string{"list", "--help"},
+			expectedOutput: "List your local Singularity cache",
+			needImage:      false,
+			exit:           0,
+		},
+		{
+			name:               "list type",
+			options:            []string{"list", "--type", "library"},
+			needImage:          true,
+			expectedOutput:     "There are 1 container file",
+			expectedEmptyCache: false,
+			exit:               0,
+		},
+		{
+			name:               "list verbose",
+			needImage:          true,
+			options:            []string{"list", "--verbose"},
+			expectedOutput:     "NAME",
+			expectedEmptyCache: false,
+			exit:               0,
+		},
+	}
+
+	// A directory where we store the image and used by separate commands
+	tempDir, imgStoreCleanup := e2e.MakeTempDir(t, "", "", "image store")
+	defer imgStoreCleanup(t)
+	imagePath := filepath.Join(tempDir, imgName)
+
+	for _, tt := range tests {
+		// Each test get its own clean cache directory
+		cacheDir, cleanup := e2e.MakeCacheDir(t, "")
+		defer cleanup(t)
+		h, err := cache.NewHandle(cacheDir)
+		if err != nil {
+			t.Fatalf("Could not create image cache handle: %v", err)
 		}
 
-		c := &cacheTests{
-			env: env,
+		if tt.needImage {
+			prepTest(t, c.env, tt.name, h, imagePath)
+		}
+
+		c.env.ImgCacheDir = cacheDir
+		c.env.RunSingularity(
+			t,
+			e2e.AsSubtest(tt.name),
+			e2e.WithCommand("cache"),
+			e2e.WithArgs(tt.options...),
+			e2e.ExpectExit(tt.exit),
+		)
+
+		if tt.needImage && tt.expectedEmptyCache {
+			ensureCacheEmpty(t, tt.name, imagePath, h)
 		}
-		t.Run("cacheClean", c.testCacheClean)
 	}
 }
 
-func (c *cacheTests) testCacheClean(t *testing.T) {
+func (c *cacheTests) testInteractiveCacheCmds(t *testing.T) {
 	tt := []struct {
-		name    string
-		options []string
-		expect  string
-		send    string
-		output  string
-		exit    int
+		name               string
+		options            []string
+		expect             string
+		send               string
+		exit               int
+		expectedEmptyCache bool // Is the cache supposed to be empty after the command is executed
 	}{
 		{
-			name:   "clean cache normal",
-			expect: "Do you want to continue? [N/y]",
-			send:   "y",
-			exit:   0,
+			name:               "clean normal confirmed",
+			options:            []string{"clean"},
+			expect:             "Do you want to continue? [N/y]",
+			send:               "y",
+			expectedEmptyCache: true,
+			exit:               0,
+		},
+		{
+			name:               "clean normal not confirmed",
+			options:            []string{"clean"},
+			expect:             "Do you want to continue? [N/y]",
+			send:               "n",
+			expectedEmptyCache: false,
+			exit:               0,
+		},
+		{
+			name:               "clean dry-run confirmed",
+			options:            []string{"clean", "--dry-run"},
+			expect:             "Do you want to continue? [N/y]",
+			send:               "y",
```

Comment:
Out of curiosity, why is `cache clean --dry-run` interactive? It should not be imo.

Reply:
It seems like a bug -> https://github.com/sylabs/singularity/issues/4225

Reply:
@WestleyK I have no idea. Since I wrote the tests, I have to admit I am even more confused than I was before. For cleaning, we now have `--dry-run`, `--force`, `--type` and `--name` and they all a different effect on the cache. A good example of something I do not understand is `--dry-run`.
```
$ singularity cache clean --dry-run
This will delete everything in your cache (containers from all sources and OCI blobs). 
Hint: You can see exactly what would be deleted by canceling and using the --dry-run option.
Do you want to continue? [N/y] y
Would remove /home/gvallee/.singularity/cache/library
Would remove /home/gvallee/.singularity/cache/oci-tmp
Would remove /home/gvallee/.singularity/cache/shub
Would remove /home/gvallee/.singularity/cache/oci
Would remove /home/gvallee/.singularity/cache/net
Would remove /home/gvallee/.singularity/cache/oras
```
which I personally do not understand how it is useful information. I am not saying that I think the developments we saw lately are bad, I just do not personally understand. 
What bothers me the most is that checking whether the cache is clean or not ends up being a non-trivial thing. And I believe it shows we still do not have good semantics behind these commands.

Reply:
@WestleyK Do you think we should open an issue? This is PR is really about e2e tests, I would like to avoid modifying anything outside of e2e.

Reply:
This one? https://github.com/sylabs/singularity/issues/4225 I just opened it.